.github/workflows/java-21.yml

```diff
@@ -0,0 +1,71 @@
+# Java 21 CI with Maven
+name: Java 21 CI with Maven
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    types: [ labeled, opened, synchronize, reopened ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: 21
+          cache: maven
+
+      - name: Build with Maven
+        run: mvn -B verify -P run-its
+
+      - name: Generate JaCoCo badge
```

Comment:
@Brijeshthummar02 it's very nice to add the coverage information at the CI process :clap: ! Could you improve the other ones as well: java-11.yml and java-17.yml?